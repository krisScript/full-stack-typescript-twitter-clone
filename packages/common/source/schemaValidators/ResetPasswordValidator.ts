import * as yup from 'yup';

interface UserValidatorType {
  password: string;
  confirmPassword: string;
}
const UserValidator = yup.object<UserValidatorType>().shape({
  password: yup
    .string()
    .trim()
    .min(12)
    .required(),
  confirmPassword: yup
    .string()
    .trim()
    .min(12)
    .oneOf([yup.ref('password')], "Passwords don't match")
    .required(),
});

export default UserValidator;
