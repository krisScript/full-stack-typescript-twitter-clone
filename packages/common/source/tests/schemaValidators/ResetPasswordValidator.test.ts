import ResetPasswordValidator from '@schemaValidators/ResetPasswordValidator';

describe('ResetPasswordValidator', (): void => {
  const password = '123456789012';
  it(`should validate successfully`, async (): Promise<void> => {
    const resetPasswordObj = { password, confirmPassword: password };
    await expect(
      ResetPasswordValidator.validate(resetPasswordObj),
    ).resolves.toBe(resetPasswordObj);
  });
  it(`should throw an errors`, async (): Promise<void> => {
    const invalidPassword = 'invalid';
    const resetPasswordObj = {
      password: invalidPassword,
      confirmPassword: invalidPassword,
    };
    await expect(
      ResetPasswordValidator.validate(resetPasswordObj),
    ).rejects.toMatchSnapshot();
  });
  it(`should throw an errors`, async (): Promise<void> => {
    const confirmPassword = 'invalid';
    const resetPasswordObj = { password, confirmPassword };
    await expect(
      ResetPasswordValidator.validate(resetPasswordObj),
    ).rejects.toMatchSnapshot();
  });
});
