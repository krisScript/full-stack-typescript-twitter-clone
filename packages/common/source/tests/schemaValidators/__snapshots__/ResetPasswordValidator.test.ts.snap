// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordValidator should throw an errors 1`] = `[ValidationError: confirmPassword must be at least 12 characters]`;

exports[`ResetPasswordValidator should throw an errors 2`] = `[ValidationError: Passwords don't match]`;
